sqlmesh/core/plan/definition.py

```diff
@@ -679,15 +694,15 @@ def _ensure_no_forward_only_revert(self) -> None:
                 )
             ):
                 raise PlanError(
-                    f"Attempted to revert to an unrevertable version of model '{name}'. Run `sqlmesh plan` again to mitigate the issue."
+                    f"Attempted to revert to an unrevertable version of model '{s_id.name}'. Run `sqlmesh plan` again to mitigate the issue."
                 )
 
     def _ensure_no_broken_references(self) -> None:
         for snapshot in self.context_diff.snapshots.values():
-            broken_references = set(self.context_diff.removed_snapshots) & snapshot.node.depends_on
+            broken_references = set(self.context_diff.removed_snapshots) & set(snapshot.parents)
             if broken_references:
                 raise PlanError(
-                    f"Removed {broken_references} are referenced in '{snapshot.name}'. Please remove broken references before proceeding."
+                    f"""Removed {', '.join(f"'{x.name}'" for x in broken_references)} are referenced in '{snapshot.name}'. Please remove broken references before proceeding."""
```

Comment:
[Nit] Can we please transform `broken_references` in a separate line? This is really hard to parse